Source/JavaScriptCore/jit/ThunkGenerators.cpp

```diff
@@ -144,6 +144,24 @@ inline void emitPointerValidation(CCallHelpers& jit, GPRReg pointerGPR, TagType
 #endif
 }
 
+MacroAssemblerCodeRef<JITThunkPtrTag> throwExceptionFromCallGenerator(VM& vm)
+{
+    CCallHelpers jit;
+
+    jit.emitFunctionPrologue();
+
+    jit.copyCalleeSavesToEntryFrameCalleeSavesBuffer(vm.topEntryFrame, GPRInfo::argumentGPR0);
+    jit.setupArguments<decltype(operationLookupExceptionHandler)>(CCallHelpers::TrustedImmPtr(&vm));
+    jit.prepareCallOperation(vm);
+    jit.move(CCallHelpers::TrustedImmPtr(tagCFunction<OperationPtrTag>(operationLookupExceptionHandler)), GPRInfo::nonArgGPR0);
```

Comment:
If you convert `isStackOverflowFrame` to `isPartiallyInitializedFrame` (which is used both by you here and by stack overflow), you can use `operationLookupExceptionHandlerFromCallerFrame` here.  It is semantically the same except for the extra ASSERTs.  However, it communicates more clearly that we intend to throw from the caller, and that the current frame is a partially initialized frame.

Reply:
Changed.

Reply:
This function can be used for non partially initialized frame case (calling a function caused an error). So we should use operationLookupExceptionHandler